### PR TITLE
ref(replays): Read from the rrweb saved config to see if replay network details is setup

### LINKED ISSUE #48733: [Replay Details] Update ReplayReader.isNetworkDetailsSetup() to read from the config breadcrumb
Follows: https://github.com/getsentry/sentry-javascript/pull/8011

### PR DESCRIPTION
**Pull request by ryan953**
Fixes https://github.com/getsentry/sentry/issues/48733

See also: https://github.com/getsentry/sentry-docs/pull/6943

### CODE REVIEW

File: static/app/views/replays/detail/network/details/onboarding.tsx [hunk 1]
```diff
@@ -121,7 +121,7 @@ export function Setup({
   showSnippet: Output;
   visibleTab: TabKey;
 }) {
-  const minVersion = '7.50.0';
+  const minVersion = '7.51.1';
```
Comment (ryan953) [inline static/app/views/replays/detail/network/details/onboarding.tsx:124]: see also https://github.com/getsentry/sentry-docs/pull/6943

File: static/app/views/replays/types.tsx [hunk 2]
```diff
@@ -155,6 +155,19 @@ export interface Highlight {
 }
 
 export type RecordingEvent = eventWithTime;
+export type RecordingOptions = customEvent<{
```
Comment (billyvg) [inline static/app/views/replays/types.tsx:158]: Hmm we should probably have this type defined in SDK.

File: static/app/utils/replays/replayReader.tsx [hunk 3]
```diff
@@ -151,11 +152,14 @@ export default class ReplayReader {
...
+  sdkConfig = memoize(() => {
+    const found = this.rrwebEvents.find(
+      event => event.type === 5 && event.data.tag === 'options'
+    ) as undefined | RecordingOptions;
+    return found?.data?.payload;
+  });
+
+  isNetworkDetailsSetup = memoize(() => {
+    return this.sdkConfig()?.networkDetailHasUrls;
+  });
```
Comment (billyvg) [inline static/app/utils/replays/replayReader.tsx:164]: Should we fallback to old method in order to support users w/ SDK < 7.51.1?

Comment (ryan953) [reply static/app/utils/replays/replayReader.tsx]: Thats a good idea, and we won't really need to bump the min version anymore. Done, and closed the related docs pr as not needed.

**Review by billyvg — APPROVED** [review]

**Pull request merged**